tests/drivers/counter/counter_basic_api/src/test_counter.c

```diff
@@ -66,8 +66,20 @@ static const char * const devices[] = {
 #if DT_HAS_NODE(DT_INST(0, arm_cmsdk_dtimer))
 	DT_LABEL(DT_INST(0, arm_cmsdk_dtimer)),
 #endif
-#ifdef DT_RTC_0_NAME
-	DT_RTC_0_NAME,
+#if DT_HAS_COMPAT(microchip_xec_timer)
```

Comment:
why are some lines `DT_HAS_COMPAT` and others `DT_HAS_NODE`?

Reply:
The DT_HAS_NODE existed before, I was just reworking the cases for DT_RTC_0_NAME getting set.

I'm sure this is due to whomever updated the test for their given driver support.

Reply:
Rework it so everything but the NRF case is the same now.